src/utils/havoc.js

```diff
@@ -210,13 +211,15 @@ class ObjectValueHavocingVisitor {
             } else {
               if (realmGenerator !== undefined) {
                 let targetDescriptor = this.getHeapInspector().getTargetIntegrityDescriptor(obj);
-                if (
-                  descriptor.writable !== targetDescriptor.writable ||
-                  descriptor.configurable !== targetDescriptor.configurable
-                ) {
-                  realmGenerator.emitDefineProperty(obj, name, descriptor);
-                } else {
-                  realmGenerator.emitPropertyAssignment(obj, name, value);
+                if (!isReactElement(obj)) {
```

Comment:
Why are React elements special? It seems like we should check for frozen or final objects instead?

Reply:
Agreed that just disabling it for React is fishy. It would only work by chance then.

I know that there’s an issue with frozen objects, the ordering with Object.freeze. Is that the problem, and not “final”?

Reply:
I updated the summary above. At this point in time, ReactElements are the only final objects that have a dedicated serialization codepath that ensures that any of their delayed properties are output together with the entire object).

Reply:
@NTillmann Yes, it seems like the same issue for Object.freeze and ordering. Any properties that are delayed are emitted way after the function call. I updated the summary to explain why ReactElements are special.